platform/src/main/java/org/hillview/storage/CassandraSSTableLoader.java

```diff
@@ -309,78 +270,77 @@ public ITable load(){
+                    // Extracting the value for each column from a single row
                     for (ColumnData cd : row) {
-                        if (cd.column().isSimple()) {
-                            cell = (Cell) cd;
-                            cellType = cell.column().cellValueType();
-                            value = cellType.getSerializer().deserialize(cell.value());
-                        } else {
-                            throw new RuntimeException("[Hillview can't convert complex data]");
+                        // Filter the column to only load the one marked true by columnToLoad[]
+                        if (columnToLoad[i]) {
+                            if (cd.column().isSimple()) {
+                                cell = (Cell) cd;
+                                cellType = cell.column().cellValueType();
+                                value = cellType.getSerializer().deserialize(cell.value());
+                            } else {
+                                // Hillview won't process the complex data
+                                throw new RuntimeException("[Hillview can't convert complex data]");
+                            }
+                            line.add(value);
                         }
-                        line[i] = value; 
                         i++;
                     }
-
-                    /* Append each line to Hillview column */
-                    for (this.currentColumn = 0; this.currentColumn < line.length; this.currentColumn++) {
-                        this.columns[this.currentColumn].append(line[this.currentColumn]);
+                    // Append each line to Hillview column
+                    for (i = 0; i < size; i++) {
+                        columns.get(i).append(line.get(i));
                     }
-                    this.currentRow++;
-
-                }else if (unfiltered instanceof RangeTombstoneMarker){
-                    /* Hillview ignores Tombstone Markers */
+                } else {
+                    // if (unfiltered instanceof RangeTombstoneMarker)
+                    // Hillview ignores Tombstone Markers
                 }
             }
         });
 
-        IColumn[] arrColumns = new IColumn[this.columns.length];
-        IMembershipSet membershipSet = null;
-        IAppendableColumn appendableColumn;
-        IColumn column;
-        for (int idx = 0; idx < this.columns.length; idx++) {
-            appendableColumn = this.columns[idx];
-            column = appendableColumn.seal();
-            if (membershipSet == null)
-                membershipSet = new FullMembershipSet(column.sizeInRows());
-            arrColumns[idx] = column;
-            assert arrColumns[idx] != null;
-        }
-
-        return new Table(arrColumns, this.ssTablePath, null);
+        for (IAppendableColumn c : columns)
+            c.seal();
+        return Linq.map(columns, e -> e);
```

Comment:
this is really useless; just return columns.

Reply:
I removed the "for" loop. However, the "Linq.map" is still needed to convert List<IAppendableColumn> to List<IColumn>

Reply:
The right solution is probably to change the API of IColumnLoader to return List<T extends IColumn>. Maybe you can try to do this.

Reply:
Okay! This will trigger some changes in other classes, such as JdbcDatabase and ParquetFileLoader